target/snitch_cluster/sw/apps/snax-xdma-memset/src/snax-xdma-memset.c

```diff
@@ -0,0 +1,126 @@
+// Copyright 2024 KU Leuven.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Yunhao Deng <yunhao.deng@kuleuven.be>
+
+#include "data.h"
+#include "snax-xdma-lib.h"
+#include "snrt.h"
+
+int main() {
+    // Set err value for checking
+    int err = 0;
+
+    // Obtain the start address of the TCDM memory
+    uint8_t *tcdm_baseaddress = (uint8_t *)snrt_l1_next();
+    uint8_t *tcdm_0 = tcdm_baseaddress;
+    uint8_t *tcdm_16 = tcdm_baseaddress + 0x4000 * sizeof(uint8_t);
+    uint8_t *tcdm_32 = tcdm_baseaddress + 0x8000 * sizeof(uint8_t);
+    uint8_t *tcdm_48 = tcdm_baseaddress + 0xc000 * sizeof(uint8_t);
+    uint8_t *tcdm_64 = tcdm_baseaddress + 0x10000 * sizeof(uint8_t);
+    uint8_t *tcdm_80 = tcdm_baseaddress + 0x14000 * sizeof(uint8_t);
+    uint8_t *tcdm_96 = tcdm_baseaddress + 0x18000 * sizeof(uint8_t);
+    uint8_t *tcdm_112 = tcdm_baseaddress + 0x1c000 * sizeof(uint8_t);
+
+    // Using xdma core only
+    if (snrt_cluster_core_idx() == snrt_cluster_compute_core_num() - 1) {
+        // The xdma core is the last compute core in the cluster
+        if (xdma_memcpy_1d(tcdm_0, tcdm_0, 0x4000 * sizeof(uint8_t)) != 0) {
+            printf("Error in xdma agu configuration\n");
+        } else {
+            printf("The xdma agu is configured\n");
+        }
+
+        uint32_t ext_param[1] = {0xFFFFFFFF};
+        if (xdma_enable_dst_ext(0, ext_param) != 0) {
+            printf("Error in enabling xdma extension 0\n");
+        } else {
+            printf("The xdma extension 0 is enabled\n");
+        }
+
+        if (xdma_disable_dst_ext(1) != 0) {
+            printf("Error in disabling xdma extension 1\n");
+        } else {
+            printf("The xdma extension 1 is disabled\n");
+        }
+
+        if (xdma_disable_dst_ext(2) != 0) {
+            printf("Error in disabling xdma extension 2\n");
+        } else {
+            printf("The xdma extension 2 is disabled\n");
+        }
+
+        int task_id = xdma_start();
+        printf(
+            "The xdma is started, setting memory region to 0xFF. The task id "
+            "is %d\n",
+            task_id);
+        xdma_wait(task_id);
+
+        printf("The xdma is finished\n");
+        // Check the data
+        for (int i = 0; i < 0x4000; i++) {
+            if (tcdm_0[i] != 0xFF) {
+                printf("Error in memset\n");
+                err = 1;
```

Comment:
I would increment the error value: `err++;` for every incorrect answer so we have information feedback on how much error is occurring. I mean when this loop breaks it will still do the `printf("The memset of 0KB - 16KB is correct\n");` right? Then you can also return without proceeding to the next to terminate simulation quick in case something is wrong.